server/src/appFactory.ts

```diff
@@ -185,6 +189,34 @@ async function appFactory(db: Repository, listoData: DirectoryData) {
     }
   });
 
+  apiRouter.get('/quickChecklist/:id', async (req, res) => {
+    try {
+      const quickChecklist = await db.getQuickChecklist(req.params.id);
+      res.json({ quickChecklist: quickChecklist, status: 200 });
+    } catch (err) {
+      console.error(`Failed to find QuickChecklist with ${req.params.id}`, err);
+      res.status(404).send(`QuickChecklist not found`);
+    }
+  });
+
+  apiRouter.post('/upsertQuickChecklist', async (req, res) => {
```

Comment:
I would take `upsert` out of the path and change it to `.put` instead of `.post` (which is usually what Upsert implies).

Also not sure how I feel about `CamelCase` in REST API paths. I would use `kebab-case` or even just use one word, e.g.

`/quick-checklist` or `quickchecklist`

Reply:
i like quick-checklist, so will go with that. ill also change the existing routes to match as they are also camel case.